azurerm/resource_arm_data_lake_store.go

```diff
@@ -65,15 +110,26 @@ func resourceArmDateLakeStoreCreate(d *schema.ResourceData, meta interface{}) er
 	location := azureRMNormalizeLocation(d.Get("location").(string))
 	resourceGroup := d.Get("resource_group_name").(string)
 	tier := d.Get("tier").(string)
+
+	encryptionState := account.EncryptionState(d.Get("encryption_state").(string))
+	encryptionType := account.EncryptionConfigType(d.Get("encryption_type").(string))
```

Comment:
Should we do a d.GetOk with this value being Optional?

Reply:
Nope, since otherwise there's no way to clear it (required when encryption is disabled)